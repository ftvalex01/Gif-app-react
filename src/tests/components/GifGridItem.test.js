// debe mostrar el componente correctamente

//shallow , wrapper , wrapper .toMatchSnapshot()

import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('GifGridItem test', () => {
    
    const title='un titulo';
    const url='https://google.com/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('debe mostar el componente correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de tener un parrafo con el title', () => {
        
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)

    });

    test('debe de tener la imagen igual al url y alt de los props',()=>{

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('debe de tener animate fadeInDownBig', () => {
        
        const div = wrapper.find('div')
        expect(div.hasClass('animate__fadeInDownBig')).toBe(true)
    });



});