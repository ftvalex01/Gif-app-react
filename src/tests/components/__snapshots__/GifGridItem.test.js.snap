// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem test debe mostar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDownBig"
>
  <img
    alt="un titulo"
    src="https://google.com/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
